File: TEST/RS/test_type.rs
struct SIMPLE
{
    uuid : String,
    bool : bool,
    int_8 : i8,
    uint_8 : u8,
    int_16 : i16,
    uint_16 : u16,
    int_32 : i32,
    uint_32 : u32,
    int_64 : i64,
    uint_64 : u64,
    float_32 : f32,
    float_64 : f64,
    string : String,
    name : String,
    date : String,
    date_time : String,
    blob : String,
    option : String,
    english_text : String,
    french_text : String,
    german_text : String,
    latin_text : String,
    spanish_text : String
}

// ~~

struct COMPOUND
{
    id : i32,
    location : String,
    name : (String, String),
    name_set : List<(String, String)>,
    phone_list : List<String>,
    email_set : List<String>,
    company_map : Map<String, String>,
    simple_date : String,
    simple_date_map : Map<(String, String), String>,
    simple_date_set : List<String>,
    simple_date_list : List<String>,
    name_set_map : Map<String, List<(String, String)>>,
    text : String,
    simple_pointer_array : Vec<Weak<SIMPLE>>
}

// ~~

struct VALUE
{
    id : i64,
    uuid : String,
    name : String,
    integer : i32,
    text : String
}

// ~~

struct VALUE_DATA
{
    uuid : String,
    value_id : i64,
    value_uuid : String,
    internal_id : i64,
    internal_uuid : String,
    internal_value_id : i64,
    internal_value_uuid : String
}

// ~~

struct INTERNAL_DATA
{
    id : i64,
    uuid : String,
    value_id : i64,
    value_uuid : String,
    first_name : String,
    last_name : String,
    is_awesome : bool
}

File: TEST/JS/test_type.js
class SIMPLE
{
    constructor(
        )
    {
        this.Uuid = null;
        this.Bool = null;
        this.Int8 = null;
        this.Uint8 = null;
        this.Int16 = null;
        this.Uint16 = null;
        this.Int32 = null;
        this.Uint32 = null;
        this.Int64 = null;
        this.Uint64 = null;
        this.Float32 = null;
        this.Float64 = null;
        this.String = null;
        this.Name = null;
        this.Date = null;
        this.DateTime = null;
        this.Blob = null;
        this.Option = null;
        this.EnglishText = null;
        this.FrenchText = null;
        this.GermanText = null;
        this.LatinText = null;
        this.SpanishText = null;
        this.Integer = null;
        this.Natural = null;
        this.Real = null;
        this.Address = null;
        this.CategorySlug = null;
        this.CategoryName = null;
        this.CategorySection = null;
    }
}

// ~~

class COMPOUND
{
    constructor(
        )
    {
        this.Id = null;
        this.Location = null;
        this.Name = null;
        this.NameSet = null;
        this.PhoneList = null;
        this.EmailSet = null;
        this.CompanyMap = null;
        this.SimpleDate = null;
        this.SimpleDateMap = null;
        this.SimpleDateSet = null;
        this.SimpleDateList = null;
        this.NameSetMap = null;
        this.Text = null;
        this.SimplePointerArray = null;
    }
}

// ~~

class VALUE
{
    constructor(
        )
    {
        this.Id = null;
        this.Uuid = null;
        this.Name = null;
        this.Integer = null;
        this.Text = null;
    }
}

// ~~

class VALUE_DATA
{
    constructor(
        )
    {
        this.Uuid = null;
        this.ValueId = null;
        this.ValueUuid = null;
        this.InternalId = null;
        this.InternalUuid = null;
        this.InternalValueId = null;
        this.InternalValueUuid = null;
    }
}

// ~~

class INTERNAL_DATA
{
    constructor(
        )
    {
        this.Id = null;
        this.Uuid = null;
        this.ValueId = null;
        this.ValueUuid = null;
        this.FirstName = null;
        this.LastName = null;
        this.IsAwesome = null;
    }
}
